API Spec update

jpengar opened the pull request:
#### What type of PR is this?

* enhancement/feature

#### What this PR does / why we need it:

As agreed in the last ICM call (https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/131399863/2025-06-04+Draft+Agenda+ICM+Minutes), this PR incorporates the proposed changes to the API spec by Telefónica to facilitate further discussion and agreement.

This PR builds on the existing Consent Info API to provide a better model for the expected functional scope, including simplified statuses, a legal basis for responses (following DPV defined legal basis), coverage of the Legitimate Interest opt-out scenario and an expiration date where applicable...

#### Which issue(s) this PR fixes:

Fixes #9
Fixes #10 
Fixes #15 
Fixes #19
Fixes #26 

#### Special notes for reviewers:

CC @HuubAppelboom @subha5h

These changes collectively improve the API's ability to handle complex consent scenarios, enhance compliance, and provide clearer documentation for developers.

**Regarding issue #11, we can discuss it further in @subha5h's PR https://github.com/camaraproject/ConsentInfo/pull/20.** That issue is NOT covered in this PR.

#### Changelog input

```
 API spec update 
```

#### Additional documentation 

N/A

jpengar commented on code/API_definitions/consent-info.yaml:
```diff
@@ -132,88 +136,111 @@ paths:
-                        consentRequired: true
-                        consentStatus: ACTIVE
+                        legalBasis: "dpv:Consent"
+                        status: PENDING
```
It sounds reasonable to me. I've addressed your proposal in https://github.com/camaraproject/ConsentInfo/pull/22/commits/48a694a58690a600a1ecf68441d097c971345e2d, along with some description improvements. Please take a look @subha5h

jpengar commented:
Hi @subha5h @shilpa-padgaonkar @bigludo7 @HuubAppelboom,
Could you please take a moment to review this PR? I believe it addresses issues #9, #10, #15, and #19.

I'd love to get it merged this week, ideally with the working group’s agreement in the upcoming ICM meeting on Wednesday (18/06).

CC @AxelNennker @sebdewet – as code owners, your review would be really appreciated as well.
Thanks a lot in advance, everyone! 🙏

shilpa-padgaonkar commented on code/API_definitions/consent-info.yaml:
```diff
@@ -3,32 +3,38 @@ info:
 
     # Introduction
 
+    The primary function of this API is to enable API Consumers to ascertain the authorization status for processing personal data before using other APIs. By providing a standardized way to check the validity of a legal basis, this API facilitates compliance with privacy regulations and promotes transparency with users. Key inputs include the specific scope(s) of access being requested, the Purp ...
```
Since this API is mainly to provide info to the API consumers as stated in the previous line, not sure how it promotes transparency with users? If I understand it right, we don't mean API consumers = users.

shilpa-padgaonkar replied:
```suggestion
    The primary function of this API is to enable API Consumers to ascertain the authorization status for processing personal data before using other APIs. Key inputs include the specific scope(s) of access being requested, the Purpose for data processing, and an identifier for the User (either explicitly provided or derived from a three-legged access token). The API responds with the validity status of the applicable Legal Basis, and, if the Legal Basis is `dpv:Consent`, can provide a URL for the API Consumer to manage it.
```

shilpa-padgaonkar commented on code/API_definitions/consent-info.yaml:
```diff
@@ -3,32 +3,38 @@ info:
     The "Camara Security and Interoperability Profile" provides details of how an API consumer requests an access token. Please refer to Identity and Consent Management (https://github.com/camaraproject/IdentityAndConsentManagement/) for the released version of the profile.
 
-    The specific authorization flows to be used will be agreed upon during the onboarding process, happening between the API consumer and the API provider, taking into account the declared purpose for accessing the API, whilst also being subject to the prevailing legal framework dictated by local legislation.
+    The specific authorization flows to be used will be agreed upon during the onboarding process, happening between the API consumer and the API provider, taking into account the declared Purpose for accessing the API, whilst also being subject to the prevailing legal framework dictated by local legislation.
```
I would not add this content here. It is already a part of relevant documents in either ICM and/or Commonalities. The more places we duplicate it, the more issues we face when we try to change some wording and then have to ensure things are consistent. A broad statement as written above asking to refer to ICM documentation is good enough IMHO.

shilpa-padgaonkar commented on code/API_definitions/consent-info.yaml:
```diff
@@ -3,32 +3,38 @@ info:
-    The specific authorization flows to be used will be agreed upon during the onboarding process, happening between the API consumer and the API provider, taking into account the declared purpose for accessing the API, whilst also being subject to the prevailing legal framework dictated by local legislation.
+    The specific authorization flows to be used will be agreed upon during the onboarding process, happening between the API consumer and the API provider, taking into account the declared Purpose for accessing the API, whilst also being subject to the prevailing legal framework dictated by local legislation.
 
     In cases where personal data is processed by the API and users can exercise their rights through mechanisms such as opt-in and/or opt-out, the use of three-legged access tokens is mandatory. This ensures that the API remains in compliance with privacy regulations, upholding the principles of transparency and user-centric privacy-by-design.
```
I would not add this content here. It is already a part of relevant documents in either ICM and/or Commonalities. The more places we duplicate it, the more issues we face when we try to change some wording and then have to ensure things are consistent. A broad statement as written above asking to refer to ICM documentation is good enough IMHO.

jpengar commented on code/API_definitions/consent-info.yaml:
```diff
@@ -264,71 +302,95 @@ components:
         - scopes
         - purpose
-        - consentRequired
+        - legalBasis
```
Thanks, Shilpa — you raise an interesting point regarding data minimization. We agree that unnecessary exposure of information should be avoided. However, we believe that including the `legalBasis` in the response brings concrete value to the API Consumer and supports meaningful interactions with users. A few reasons:

- Clarity for user interaction: When the processing is not allowed (statusValidForProcessing = false), knowing why—e.g. because a dpv:Consent has expired vs. a user having opted out of dpv:LegitimateInterest—is essential. In the former, the API Consumer can re-engage the user and ask for consent again; in the latter, it should not. Without legalBasis, this distinction is lost.
- Interoperability: If legalBasis is optional, different API Providers might omit it. This results in inconsistent behavior for the API Consumer, who cannot implement a single logic path. We believe this negatively impacts the API’s usability and interoperability.
- Alignment with onboarding practices: In TMF-based onboarding, API Providers already communicate the applicable legal basis for specific data scopes. This API just exposes that same information at runtime, when it's contextually valuable. This enhances—not contradicts—transparency and predictability.

shilpa-padgaonkar replied:
@jpengar You are right. Since we are anyways making this info available as a part of onboarding, would make sense to be consistent..

jpengar commented on code/API_definitions/consent-info.yaml:
```diff
@@ -3,32 +3,38 @@ info:
     * **Consent**: An explicit opt-in action that the User takes to allow processing of Personal Data. Consent grants the API Consumer access to a set of scopes related to the User for a specific Purpose.
     * **Purpose**: The reason for which Personal Data will be processed by an API Consumer. CAMARA defines a standard set of Purposes which can be used by API Consumers to specify the reason for their intended Personal Data processing.
+    * **Scope**: A string representing the specific access rights or actions an API Consumer requests from the User for their data (e.g., "location-verification:verify"). A request can contain multiple scopes.
+    * **Legal Basis**: The justification under data protection law for processing Personal Data. This API uses terms from the W3C Data Privacy Vocabulary (DPV), such as `dpv:Consent`, `dpv:LegitimateInterest`, `dpv:Contract`, etc., to indicate the relevant legal ground.
```
Fixed in https://github.com/camaraproject/ConsentInfo/pull/22/commits/4e6a5286872d28834bbf00a2f3a7803c8b1109e5

shilpa-padgaonkar commented on code/API_definitions/consent-info.yaml:
```diff
@@ -264,71 +304,95 @@ components:
+        statusReason:
           type: string
           enum:
             - PENDING
```
Do we need REQUESTED and UNKNOWN instead of pending?
REQUESTED would mean, end user has been asked, but the end user is yet to respond.

UNKNOWN is when the end user has not been asked yet?

jpengar replied:
'PENDING' is defined as follows: The specified legal basis is not yet established or fully validated for the requested data processing. For example, the user has not yet provided consent under the 'dpv:Consent' legal basis.' This is not specific to dpv:Consent, so it has been defined in a generic way to cover all possible statuses for the different legal bases where they are not yet fully ready or established. If you look at the W3C DPV, you will see that it defines many statuses that could potentially map here. I would leave it as it is.

shilpa-padgaonkar replied:
@jpengar Yes, I am aware that we don't follow W3C DPV yet for consent status, but I would like to keep it as close as possible so that the transition is easier.

AxelNennker commented on code/API_definitions/consent-info.yaml:
```diff
@@ -264,71 +304,95 @@ components:
           format: url
           description: |
-            URL where the User can provide the necessary Consent. This field is only present if requested by the API Consumer and if `privacyStatus[*].consentRequired` is `true` and its `privacyStatus[*].consentStatus` is `PENDING` or `EXPIRED`.
+            URL where the User can provide the necessary Consent. This field is only present if requested by the API Consumer and if `statusInfo[*].legalBasis` is `dpv:Consent`, `statusInfo[*].statusValidForProcessing` is `false`, and `statusInfo[*].statusReason` is `PENDING` or `EXPIRED`.
```
```suggestion
            URL where the User can provide the necessary Consent. This field is only present if requested by the API Consumer and if `statusInfo[*].legalBasis` is `dpv:Consent`, `statusInfo[*].statusValidForProcessing` is `false`, and `statusInfo[*].statusReason` is `REQUESTED` or `EXPIRED`.
```

AxelNennker commented on code/API_definitions/consent-info.yaml:
```diff
@@ -264,71 +304,95 @@ components:
+        statusReason:
           type: string
           enum:
             - PENDING
```
```suggestion
            - REQUESTED
```

AxelNennker reviewed (changes requested):
I think "REQUESTED" is closer to the data privacy vocabulary than "PENDING".

jpengar commented:
> I think "REQUESTED" is closer to the data privacy vocabulary than "PENDING".

@AxelNennker @shilpa-padgaonkar I think 'requested' is as far or close as 'pending' to the data privacy vocabulary. Therefore, I don't see any reason to replace "PENDING" with "REQUESTED". How is "REQUESTED" closer to DPV? 'Requested' is also a very specific status, mapping pretty much to a situation where consent has been requested but is not yet provided. However, first of all, the 'PENDING' status and the others are not intended just for the 'Consent' legal basis, but for the rest of them too. DPV defines many other, very different statuses that simply don't fit with 'REQUESTED'. For example, with regard to consent, consent could not yet have been granted yet, and additionally not requested or prompted to the user. And if we consider other legal bases, it simply does not fit. What would the meaning of 'REQUESTED' be for a contract, legitimate interest or legal obligation?

Here are some specific examples that do not map to REQUESTED:

dpv:LegalObligationPending
DPV:LegitimateInterestUninformed
dpv:PublicInterestPending
dpv:ContractualClauseNotFulfilled
etc, etc, etc

Ref: https://w3c.github.io/dpv/2.1/dpv/#vocab-legal-basis-consent

P.S. Rather than simply blocking the PR, I would like to discuss this in today's meeting.

jpengar commented:
> @jpengar Just curious: What do statuses "pending" and "expired" mean when legal base is legitimate interest (and the enduser has option to opt-out)?

You're right to ask — but in the case of legitimate interest where the end-user has the option to opt out, the statuses pending and expired are not relevant. As documented in the API spec, if the user opts out, the status will be REVOKED, which has nothing to do with either "pending" or "expired".

However, for other legal bases, such as dpv:LegalObligationPending, dpv:LegitimateInterestUninformed, dpv:PublicInterestPending, or dpv:ContractualClauseNotFulfilled, these describe situations where the legal basis is not fully established or is awaiting confirmation. These can justifiably map to a generic PENDING status. Mapping them to REQUESTED wouldn't make much sense in those cases, IMHO.

AxelNennker commented on code/API_definitions/consent-info.yaml:
```diff
@@ -3,32 +3,40 @@ info:
-    * Provides the current Consent status: If Consent is required, the API returns the current status (e.g., active, revoked, expired...) so the API Consumer can understand the current User's consent status.
-    * Offers a Consent Capture URL: If the user has not provided the necessary Consent, the API supplies an API Provider's Consent capture URL where the user can easily complete the Consent process.
+    * Indicates the applicable Legal Basis: Indicates which legal basis (e.g., `dpv:Consent`, `dpv:LegitimateInterest`) is required for the requested scopes, Purpose, and API Consumer according to relevant legal frameworks.
+    * Provides the data processing validity status and reason: If a legal basis is required, the API returns `statusValidForProcessing`, a boolean flag that is `true` if processing is permitted. If the flag is `false`, a `statusReason` field (with values like `PENDING`, `REVOKED`, `EXPIRED`) is also returned to explain why.
```
Additionally, a sentence that begins with an if-clause and also ends with an if-clause is hard to read.

> If a legal basis is required, the API returns `statusValidForProcessing`, a boolean flag that is `true` if processing is permitted.

Isn't there always a requirement for a legal basis? 

I can't make a suggestion because I am not sure I understand this

jpengar commented on code/API_definitions/consent-info.yaml:
```diff
@@ -3,32 +3,40 @@ info:
-    * Offers a Consent Capture URL: If the user has not provided the necessary Consent, the API supplies an API Provider's Consent capture URL where the user can easily complete the Consent process.
+    * Indicates the applicable Legal Basis: Indicates which legal basis (e.g., `dpv:Consent`, `dpv:LegitimateInterest`) is required for the requested scopes, Purpose, and API Consumer according to relevant legal frameworks.
+    * Provides the data processing validity status and reason: If a legal basis is required, the API returns `statusValidForProcessing`, a boolean flag that is `true` if processing is permitted. If the flag is `false`, a `statusReason` field (with values like `PENDING`, `REVOKED`, `EXPIRED`) is also returned to explain why.
+    * Offers a Consent Capture URL: This functionality is specific to the `dpv:Consent` legal basis. If the API Consumer sets `requestCaptureUrl` to `true`, and if a user needs to provide or renew their Consent (e.g., the `statusReason` is `PENDING` or `EXPIRED`), the API supplies the API Provider's Consent capture URL.
```
It's OK as it is IMHO.

jpengar commented:
> Please remind me: Why was "legal basis" introduced in in this PR and the ConsentInfo API.
> 
> I have the impression that we have either "consent" or "legitimate interest" in 99% of our cases. I agree that it is nice to have an API that is abstract enough to cover all the use cases globally.
> 
> Wouldn't it be simpler to have a base version of the API that handles consent and opt-out, and deal with the complicated legal bases in the next version if that is really needed?
> 
> The current proposal feels too complicated. I was a DPO for a few years of a subsidiary of DT and still I think I need this PR being reviewed by our current privacy counsels.

https://github.com/camaraproject/ConsentInfo/issues/10#issuecomment-2898318006 @HuubAppelboom 
https://github.com/camaraproject/ConsentInfo/issues/10#issuecomment-2911900657 @bigludo7 
https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/131399863/2025-06-04+Draft+Agenda+ICM+Minutes

bigludo7 commented:
@AxelNennker If we want to manage consent & opt-out as you mentioned (and I agree on this requirement) it means that we have to manage 2 legal base. So better to introduce legal base in the API (as this is a legal wording) and as @shilpa-padgaonkar mentioned the legal base is communicated to the app developer during the onboarding.

Now if we agree to manage this legal base the question is : do we restrict its value to the 2 ones that we'll use or do we have an enum with all 6 values? both can raise question depending on developer knowledge - I tend to think that having all value is easier to explain & perhaps can be used in some corner case.... but no a hard point for me if you prefer to restrict.

HuubAppelboom commented:
@AxelNennker It would be a bit strange to leave out the other 4 legal basis values (next to consent and legitimate interest), if you ask me, it alsmost seems if we would be trying to restrict some business, which is not the case.  

Also there may be edge cases that you may want to support as well:
- performance of a contract - for example to use number verify when onboarding a business customer in your own Telco operation, in case you need to be able to contact that customer by phone in case of an emergency
- vital interest - for example you may use the KYC Fill In API or a location API as a specification to connect to the Telco's for providing information to the organisation running the 911 emergency services in your country.   It could also become relevant in case of future health care services
- legal obligation - depending on legislation, this could also apply to the above case of emergency services
- public interest or official authority - in case for example police are involved, and they want to know the IMEI number of a specific MSISDN in a fraud investigation, etc.

The CAMARA API's can also be applied in some of the above edge cases, so why would we exclude these ??

sebdewet commented on code/API_definitions/consent-info.yaml:
```diff
@@ -130,90 +138,120 @@ paths:
-                        consentStatus: ACTIVE
+                        legalBasis: "dpv:Consent"
+                        statusValidForProcessing: false
+                        statusReason: PENDING
```
I think it depends on what the operator has the right to display in legal terms. Is it appropriate to display that the operator has made the request, and that it is pending validation by the end-user, or that it is by the end-user’s hand, without having been expressly required by the operator

HuubAppelboom commented:
Please note that the right to object also exsist with the legal basis PublicInterest. Not sure whether you would do this through an opt-out mechanism, though. You would probably have to object with the API Consumer.

A table filled with n/a is I think a bit useless, you can also leave the n/a columns out.

jpengar commented:
> @jpengar I just noticed that I did not reply to one of your earlier question - "How is "REQUESTED" closer to DPV? REQUESTED is one of the consent statuses listed here https://w3c.github.io/dpv/2.1/dpv/#vocab-legal-basis-consent-status under dpv:ConsentStatusInvalidForProcessing (EXPIRED AND REVOKED are also listed).
> 
> How about keeping both PENDING (which may be more applicable to other legal base like contract) and REQUESTED?

@shilpa-padgaonkar Thank you again for your valuable feedback and for driving this discussion.

I still hold that we must keep the PENDING status. Its role as a generic status is crucial for the API's design, as it needs to apply to situations across all supported legal bases (e.g., dpv:Contract, dpv:LegitimateInterest), where a specific term like REQUESTED would not be semantically correct.

However, I can see the potential value in providing more granular information for the dpv:Consent lifecycle, as you suggested. Therefore, I would be open to supporting the addition of REQUESTED as a new status that coexists with PENDING, but does not replace it. We could define their specific uses as follows:

- REQUESTED: This status could be used when the legalBasis is dpv:Consent and a consent prompt has been actively sent to the user, but they have not yet provided a response.
- PENDING: This status would remain for all other scenarios where a legal basis is not yet established. This would include cases under dpv:Consent where the process to obtain consent has not yet been initiated, as well as for any other legal basis like dpv:Contract or dpv:LegitimateInterest that is not yet validated.

The key question then becomes: is the distinction between a consent that has not been initiated (PENDING) and a consent that has been prompted but is awaiting a user response (REQUESTED) relevant enough for the API Consumer to justify adding a separate status?

If the consensus is that this added granularity provides significant value, then I would agree with this approach.

What do you think?

jpengar commented:
> What does the API return, if the legal basis allows opt-out but the user did not (yet) opt out.
> I assume READY_FOR_PROCESSING but can the API consumer know that opt-out is a legal right of the end-user?
> I think, if the API returns the legal status, then it should also say whether opt-is a legal right or not.

@AxelNennker  There is an example included in the API Spec for that scenario:

```yaml
READY_FOR_PROCESSING:
  summary: Ready for processing
  description: |
    Data processing is allowed for the requested scope(s) and Purpose.
  value:
    statusInfo:
      - scopes:
          - "number-verification:verify"
        purpose: "dpv:FraudPreventionAndDetection"
        legalBasis: "dpv:LegitimateInterest"
        statusValidForProcessing: true
```
If a user has the right to opt-out but has not yet done so, the API would return a status that is valid for processing (e.g., statusValidForProcessing: true), assuming all other conditions under the applicable law are met. If the user later exercises their right to opt-out, the API would then return statusValidForProcessing: false with a statusReason of REVOKED, as the specification defines this state for a "successful objection (opt-out) to processing based on legitimate interest".

However, I believe that the API specification itself should not declare whether an opt-out is a legal right for the end-user. My reasoning is that this right is entirely dependent on the specific legal framework applicable to the API Provider and the user, which can vary significantly between jurisdictions.

**I think the API's scope should be intentionally limited to reporting the current processing status, not to serve as a source of legal information. Forcing the API to declare user rights would introduce significant complexity and dependencies on various laws, which is a responsibility that should lie with the API Provider, not the API specification. The API simply provides a standardized way to communicate the outcome of that provider-side legal assessment.**

On a related note, and in line with some of Huub's recent comments, this is also why I am hesitant to create a detailed state-transition table within the spec. Such a table would be highly susceptible to error and would also be dependent on each applicable law. The API Provider is the one who must resolve this logic internally.

In summary, I believe we should keep the API focused on its core function and avoid overcomplicating it with legal interpretations. We should leave the implementation of jurisdiction-specific rules to the API Provider and not make the progress of the API specification contingent on defining them.

jpengar commented:
With the M3 milestone expected by the end of this week (https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/137265620/2025-06-19+TSC+Minutes) and the existing API design in this PR has been already approved by most of the active working group participants, do you think we can move on once we reach a conclusion on the 'REQUESTED' status thingy? @shilpa-padgaonkar @AxelNennker

AxelNennker commented:
> Please note that the right to object also exsist with the legal basis PublicInterest. Not sure whether you would do this through an opt-out mechanism, though. You would probably have to object with the API Consumer.
> 
> A table filled with n/a is I think a bit useless, you can also leave the n/a columns out.

I created the table mostly for this discussion here and also for my understanding what an API consumer can expect.
As a developer at the API consumer is would like to have a list of the potential return values of the ConsentInfo API.

if ("dpv:consent".equals(legalBasis)) then 
  if ("dpv:REQUESTED".equals(status) then
    do-something
...
else
   fail unsupported ConsentInfo response

The developer is going to implement the combinations of legalBasis and status we defined in this API spec.
The API consumer must be able to determine what to do next, what to show the end-user next, tell the end-user what they might need to do. Maybe the law requires the API consumer to tell the end-user that they have the right to opt-out.

jpengar commented:
> > REQUESTED: This status could be used when the legalBasis is dpv:Consent and a consent prompt has been actively sent to the user, but they have not yet provided a response.
> > PENDING: This status would remain for all other scenarios where a legal basis is not yet established. This would include cases under dpv:Consent where the process to obtain consent has not yet been initiated, as well as for any other legal basis like dpv:Contract or dpv:LegitimateInterest that is not yet validated.
> 
> Maybe PENDING is more like a dpv:EntityUninformed? https://w3c.github.io/dpv/2.0/dpv/#vocab-context-status

PENDING describes the state of a legal basis (e.g., consent not yet given, contract not yet signed). It doesn't mean the user is uninformed. A user can be fully aware of data processing, yet the legal basis is still PENDING.

On the other hand, dpv:EntityUninformed specifically means the user hasn't been informed.

"Cherry-picking" specific DPV states like dpv:EntityUninformed for our API's statusReason is problematic. It risks inconsistency by ignoring other relevant DPV states and overcomplicates the API for consumers. Our current approach, using a concise set of generic states like PENDING, REVOKED, and EXPIRED, is more robust. This abstracts the DPV lifecycle complexity (which the API provider manages) and offers a clear, actionable response to API consumers.

shilpa-padgaonkar commented:
> > @jpengar I just noticed that I did not reply to one of your earlier question - "How is "REQUESTED" closer to DPV? REQUESTED is one of the consent statuses listed here https://w3c.github.io/dpv/2.1/dpv/#vocab-legal-basis-consent-status under dpv:ConsentStatusInvalidForProcessing (EXPIRED AND REVOKED are also listed).
> > How about keeping both PENDING (which may be more applicable to other legal base like contract) and REQUESTED?
> 
> @shilpa-padgaonkar Thank you again for your valuable feedback and for driving this discussion.
> 
> I still hold that we must keep the PENDING status. Its role as a generic status is crucial for the API's design, as it needs to apply to situations across all supported legal bases (e.g., dpv:Contract, dpv:LegitimateInterest), where a specific term like REQUESTED would not be semantically correct.
> 
> However, I can see the potential value in providing more granular information for the dpv:Consent lifecycle, as you suggested. Therefore, I would be open to supporting the addition of REQUESTED as a new status that coexists with PENDING, but does not replace it. We could define their specific uses as follows:
> 
> * REQUESTED: This status could be used when the legalBasis is dpv:Consent and a consent prompt has been actively sent to the user, but they have not yet provided a response.
> * PENDING: This status would remain for all other scenarios where a legal basis is not yet established. This would include cases under dpv:Consent where the process to obtain consent has not yet been initiated, as well as for any other legal basis like dpv:Contract or dpv:LegitimateInterest that is not yet validated.
> 
> The key question then becomes: is the distinction between a consent that has not been initiated (PENDING) and a consent that has been prompted but is awaiting a user response (REQUESTED) relevant enough for the API Consumer to justify adding a separate status?
> 
> If the consensus is that this added granularity provides significant value, then I would agree with this approach.
> 
> What do you think?

Thanks @jpengar for your comment and for extending the existing status values with the status "REQUESTED". As I also stated earlier, I don't have a problem to keep the status "PENDING".   
One other small point, if the API provider is not able to know immediately the reason for invalidity of processing,  for whatever reason,  IMHO the status "UNKNOWN" will be a good addition.

Other than that I am fine with the rest.

jpengar commented:
> With the M3 milestone expected by the end of this week (https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/137265620/2025-06-19+TSC+Minutes) and the existing API design in this PR has been already approved by most of the active working group participants, do you think we can move on once we reach a conclusion on the 'REQUESTED' status thingy? @shilpa-padgaonkar @AxelNennker @sebdewet

With the approval of @subha5h @shilpa-padgaonkar @bigludo7 and @HuubAppelboom and of course also Telefónica (as PR owner), with a general consensus among most of the active WG participants, do you think we can move forward with the PR content for this first API version? As previously mentioned, the M3 milestone is scheduled for the end of this week and time is running out... @sebdewet @AxelNennker Could you please review it again after the latest commit? Thank you so much in advance.

AxelNennker commented:
to @shilpa-padgaonkar 's comment on data-minimization @jpengar answered that the API consumer needs to destinguish whether they need to know whether the should ask for consent again and in the case of opt-out they should not ask again, and that therefore legal basis is needed in the response.

Why not use status "ConsentRefused" meaning that the API consumer should not ask to opt-in again.
No need for legal basis in the response.

I think it no cherry picking to take from DPV what the API needs. That we allow one status does not mean that all of DPV is in the API. But I think it is good standards practice to re-use existing standards and not define something new that is hard to define.

AxelNennker commented:
My review is this PR https://github.com/AxelNennker/ConsentInfo/pull/1 which does not need legalBasis and provides the same functionality as the API spec in this PR.

AxelNennker commented:
@sebdewet just LGTM-ed https://github.com/AxelNennker/ConsentInfo/pull/1#pullrequestreview-2957329160

@jpengar please consider copying that API spec to #22

diegogonmar commented:
Hello,
To clarify the history of this topic, the decision to add the legal basis was a working group consensus reached during the June 4th meeting, as documented in the minutes. The group tasked @jpengar with creating a PR to implement this agreement.
At the subsequent meeting on June 18th, the group reaffirmed this by agreeing to notify the TSC of the scope change, to which the TSC raised no objections. This served as a final confirmation of the decision.
Therefore, this PR #22 is the direct implementation of a consensus already achieved. We believe it is not proper procedure to reopen a settled group decision within the PR that is meant to execute it.
@AxelNennker, while any member is free to disagree or change their mind, attempting to reverse the consensus via an alternative PR is procedurally incorrect. That alternative PR lacks reviews from the key participants and companies (@HuubAppelboom, @jpengar, @bigludo7, @subha5h) who validated the original agreement, so their approval cannot be transferred.
The correct path forward would be to open a new issue to formally re-evaluate the group's decision. This would ensure all stakeholders who approved the original change are included in the new discussion. We can dedicate time to this at our next meeting on July 2nd and amend the decision if the group agrees.
As a code owner, @AxelNennker you can block the merge of this PR, and we acknowledge this. If this means that we will not meet the M3 milestone, we will have to accept that outcome. For now I don't see any real risk to not have the API in Fall25, and it's more important to clarify all outstanding points and have a Release Candidate we're all comfortable with than to rush to meet M3 and potentially have to revisit discussions later.

Finally, we want to be clear that we included the legal basis field to comply with the group's action point, but it was not our original proposal. We are flexible on this matter; while we support the consensus, it is not an indispensable requirement for Telefonica, and we will not oppose a formal change in decision.

AxelNennker commented:
Please also see https://github.com/AxelNennker/ConsentInfo/pull/1#issuecomment-3013496882

jpengar commented:
Hi team,

Just a quick update: I've applied the agreed-upon changes to the API, making it LegalBasis-agnostic. See https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/152338712/2025-07-02+Draft+Agenda+ICM+Minutes

Specifically, I've:

- Removed the LegalBasis from the API response.
- Updated the API documentation and design to reflect this agnostic approach.
- Slightly simplified the info description for a more developer-friendly experience.
- Incorporated the necessary changes to update the API endpoint URL from /verify to /retrieve as per PR #27.

Unless there are any showstoppers, this PR now implements our last agreed-upon changes. As you know, we have a [hard deadline from the TSC](https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/152338445/2025-07-03+TSC+Minutes) next Tuesday, July 8th, to meet the Release Management milestone. I kindly ask you to review and approve the PR so we can merge it and generate the release on time.

Any fine-tuning of descriptions or minor adjustments can definitely be handled in a new issue and PR, and we can incorporate those before the API's public release.

Thanks for your prompt attention to this!

CC @HuubAppelboom @AxelNennker @sebdewet

jpengar commented:
@sebdewet @AxelNennker Guys, I would be very grateful if you could provide your review. Today is the deadline for the release PR to be ready and this PR requires at least one code owner's approval. Thank you so much in advance.

sebdewet commented on code/API_definitions/consent-info.yaml:
```diff
@@ -2,33 +2,35 @@ openapi: 3.0.3
     # API Functionality
 
-    This API enables an API Consumer to determine whether user Consent is required, in the applicable legislation, for the requested actions based on the provide scope(s) and Purpose. NOTE: It may be required more than one User Consent for the requested scope(s) and Purpose (e.g. when the requested scopes are related to multiple APIs).
+    This API enables the API Consumer making the request to determine whether its data processing is permitted for a given User, scope(s) and Purpose.
```
```suggestion
    This API enables the API consumer to determine whether their data processing is permitted for a given User, scope(s) and Purpose.
```

jpengar replied:
I will apply your suggestion here.

jpengar commented on code/API_definitions/consent-info.yaml:
```diff
@@ -2,33 +2,35 @@ openapi: 3.0.3
   description: |
-
-    The Consent Info API allows the API Consumer to verify the Consent status of a particular User for that API Consumer and for the requested scope(s) and Purpose. If the Consent is required for the requested scope(s) and Purpose, but has not been granted by the User, the API can also provide a URL pointing to the API Provider's Consent capture channel, enabling the API Consumer to request Conse ...
+    The Consent Info API allows API Consumers to easily validate whether they have the necessary permissions to process User's personal data for a specific Purpose before using other CAMARA APIs. It provides a simple `true`/`false` response and, when applicable, a URL to direct the User to manage their Consent.
```
I really think 'simple' is the better option here. In any case, though, this is not really relevant for the release. I will leave it as it is for now, and the text can be fine-tuned as requested after the pre-release and before the public release.

jpengar commented on code/API_definitions/consent-info.yaml:
```diff
@@ -2,33 +2,35 @@ openapi: 3.0.3
 
     # Introduction
 
+    This API facilitates compliance with privacy regulations and promotes transparency with Users by providing a standardized way to check whether, for example, the User has given their Consent or opted out of data processing. Key inputs include the specific scope(s) of access being requested (i.e. the CAMARA API(s) operations to which the data processing refers), the Purpose for data processing, ...
```
The API Consumer manages the API Provider's consent capture URL, presenting it to the user at the most opportune time and place. The user then authenticates with the API provider through that URL within the provider's secure environment. I don't see the need for a change.
